overlord/ifacestate/ifacemgr.go

```diff
@@ -64,26 +64,35 @@ func Manager(s *state.State) (*InterfaceManager, error) {
 // Connect returns a set of tasks for connecting an interface.
 //
 func Connect(s *state.State, plugSnap, plugName, slotSnap, slotName string) (*state.TaskSet, error) {
-	// TODO: Store the intent-to-connect in the state so that we automatically
-	// try to reconnect on reboot (reconnection can fail or can connect with
-	// different parameters so we cannot store the actual connection details).
+	plugRef := interfaces.PlugRef{Snap: plugSnap, Name: plugName}
+	slotRef := interfaces.SlotRef{Snap: slotSnap, Name: slotName}
+	// Store the intent to connect
+	intents := getIntents(s)
+	intents.Add(Intent{Plug: plugRef, Slot: slotRef})
+	setIntents(s, intents)
```

Comment:
I'm not sure if this answers your question but on Init() I'd like to load intents, add all snaps to the repository and spawn all the connect tasks for each of those. This will at least try to re-construct back the same state we were left off before reboot.

There is still some more work to be done for across reboots and this will be more and more interesting as we add hooks which can make each reboot rebuild the same set of intents to a different set of connections and their parameters (because hooks can alter attributes and can also fail).

Currently the actual side-effects of each connection are persistent across reboots (config for apparmor/seccomp/dbus/udev) and they are made effective even before snapd starts up (with systemd units). I wonder if given this fact we should simply persist connections directly.

I went with persistent intents vs persistent connections because I think we will move away from that model as it cannot support dynamic hooks completely.